wpilibj/src/test/java/edu/wpi/first/wpilibj/controller/SampleTrajectory.java

```diff
@@ -0,0 +1,252 @@
+      new double[]{0.020000, 23.691381, 20.187367, 5.824061, 3.448428},
+      new double[]{0.020000, 23.751480, 20.157685, 5.824835, 3.351488},
+      new double[]{0.020000, 23.809857, 20.128912, 5.825653, 3.254159},
+      new double[]{0.020000, 23.866505, 20.101049, 5.826508, 3.156454},
+      new double[]{0.020000, 23.921416, 20.074100, 5.827394, 3.058384},
+      new double[]{0.020000, 23.974584, 20.048066, 5.828306, 2.959959},
+      new double[]{0.020000, 24.026000, 20.022949, 5.829238, 2.861192},
+      new double[]{0.020000, 24.075659, 19.998747, 5.830185, 2.762095},
+      new double[]{0.020000, 24.123552, 19.975463, 5.831141, 2.662678},
+      new double[]{0.020000, 24.169673, 19.953095, 5.832102, 2.562954},
+      new double[]{0.020000, 24.214015, 19.931642, 5.833062, 2.462933},
+      new double[]{0.020000, 24.256571, 19.911104, 5.834018, 2.362628},
+      new double[]{0.020000, 24.297334, 19.891479, 5.834964, 2.262050},
+      new double[]{0.020000, 24.336297, 19.872765, 5.835896, 2.161210},
+      new double[]{0.020000, 24.373454, 19.854961, 5.836811, 2.060121},
+      new double[]{0.020000, 24.408799, 19.838064, 5.837704, 1.958794},
+      new double[]{0.020000, 24.442325, 19.822073, 5.838571, 1.857241},
+      new double[]{0.020000, 24.474028, 19.806986, 5.839410, 1.755473},
+      new double[]{0.020000, 24.503900, 19.792799, 5.840217, 1.653502},
+      new double[]{0.020000, 24.531937, 19.779511, 5.840989, 1.551341},
+      new double[]{0.020000, 24.558134, 19.767119, 5.841723, 1.449000},
+      new double[]{0.020000, 24.582486, 19.755622, 5.842416, 1.346491},
+      new double[]{0.020000, 24.604989, 19.745016, 5.843065, 1.243828},
+      new double[]{0.020000, 24.625637, 19.735299, 5.843669, 1.141020},
+      new double[]{0.020000, 24.644428, 19.726470, 5.844225, 1.038081},
+      new double[]{0.020000, 24.661358, 19.718527, 5.844731, 0.935021},
+      new double[]{0.020000, 24.676423, 19.711467, 5.845185, 0.831854},
+      new double[]{0.020000, 24.689620, 19.705290, 5.845587, 0.728591},
+      new double[]{0.020000, 24.700948, 19.699992, 5.845934, 0.625243},
+      new double[]{0.020000, 24.710403, 19.695575, 5.846225, 0.521823},
+      new double[]{0.020000, 24.717984, 19.692035, 5.846460, 0.418343},
+      new double[]{0.020000, 24.723690, 19.689372, 5.846637, 0.314815},
+      new double[]{0.020000, 24.727518, 19.687586, 5.846757, 0.211250},
+      new double[]{0.020000, 24.729614, 19.686608, 5.846822, 0.115618},
+      new double[]{0.020000, 24.730433, 19.686226, 5.846848, 0.045199},
+      new double[]{0.020000, 24.730602, 19.686148, 5.846853, 0.009310},
+      new double[]{0.020000, 24.730602, 19.686148, 5.846853, 0.000000}
+  };
+
+  private SampleTrajectory() {
+  }
+
+  public static synchronized SampleTrajectory getInstance() {
+    if (m_instance == null) {
+      m_instance = new SampleTrajectory();
+    }
+    return m_instance;
+  }
+
+  @SuppressWarnings("PMD.MethodReturnsInternalArray")
```

Comment:
But PMD is correct here!  Even though the array is final, its elements can be modified outside of the class because of this method.  As PMD recommends, we need to return a copy of the array or return an ImmutableList instead.